File: 06/coins/coins.py
from cs50 import get_float

# Return a minimum number of coins to be returned as change to the user

# SOLUTION

change = get_float("Change owed: ")
coins = 0

# Don't accept change owed thats negative
while (change < 0):
    change = get_float("Change owed: ")

# Round change by 100
rounded = change * 100

# Find the number of coins for the rounded change
while (rounded != 0):
    if (rounded >= 25):
        rounded = rounded - 25
        coins = coins + 1
    elif(rounded < 25 and rounded >= 10):
        rounded = rounded - 10
        coins = coins + 1
    elif(rounded < 10 and rounded >= 5):
        rounded = rounded - 5
        coins = coins + 1
    elif(rounded < 5 and rounded >= 1):
        rounded = rounded - 1
        coins = coins + 1
    elif(change < 0.01):
        break

print(coins)

File: 01/coins/coins.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

// Return a minimum number of coins to be returned as change to the user

// SOLUTION

int main(void)
{
    //Get user input
    float change;
    do
    {
        change = get_float("Change owed:\n");
    } while (change < 0);

    //Convert dollars into cents
    int cents = round(change * 100);

    int coins = 0;
    int newRemainder = cents;

    //Count all possible 25 cents
    while (newRemainder >= 25)
    {
        newRemainder = newRemainder - 25;
        coins = coins + 1;
    }

    //Count all possible 10 cents
    while (newRemainder >= 10)
    {
        newRemainder = newRemainder - 10;
        coins = coins + 1;
    }

    //Count all possible 5 cents
    while (newRemainder >= 5)
    {
        newRemainder = newRemainder - 5;
        coins = coins + 1;
    }

    //Count all possible 1 cents
    while (newRemainder >= 1)
    {
        newRemainder = newRemainder - 1;
        coins = coins + 1;
    }

    printf("%i", coins);
}
